near-sdk/src/store/lazy/lazy_option.rs

```diff
@@ -0,0 +1,163 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use once_cell::unsync::OnceCell;
+
+use super::{load_and_deserialize, serialize_and_store};
+use crate::env;
+use crate::utils::{CacheEntry, EntryState};
+use crate::IntoStorageKey;
+
+/// An persistent lazily loaded option, that stores a `value` in the storage when `Some(value)`
+/// is set, and not when `None` is set. `LazyOption` also [`Deref`]s into [`Option`] so we get
+/// all its APIs for free.
+///
+/// This will only write to the underlying store if the value has changed, and will only read the
+/// existing value from storage once.
+///
+/// # Examples
+/// ```
+/// use near_sdk::store::LazyOption;
+///
+///# near_sdk::test_utils::test_env::setup();
+/// let mut a = LazyOption::new(b"a", None);
+/// assert!(a.is_none());
+///
+/// *a = Some("new value".to_owned());
+/// assert_eq!(a.get(), &Some("new value".to_owned()));
+///
+/// // Using Option::replace:
+/// let old_str = a.replace("new new value".to_owned());
+/// assert_eq!(old_str, Some("new value".to_owned()));
+/// assert_eq!(a.get(), &Some("new new value".to_owned()));
+/// ```
+/// [`Option`]: core::option::Option
+/// [`Deref`]: core::ops::Deref
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct LazyOption<T>
+where
+    T: BorshSerialize,
+{
+    /// Key bytes to index the contract's storage.
+    storage_key: Vec<u8>,
```

Comment:
@austinabell do you think we should start with Box<[u8]> here to be consitent with new vec (I think you did a switch there?)

Reply:
ah, yeah, I saw this and forgot to comment the same. Similar to [this commit on the other PR](https://github.com/near/near-sdk-rs/pull/402/commits/032a968a2a01c3ede88e2be93554887bf9388485) based on Aleksey's review it probably makes sense to make this change.

I added this in #387 the other day to change it more generally